src/dmd/expressionsem.d

```diff
@@ -8684,9 +8750,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
         // for static alias this: https://issues.dlang.org/show_bug.cgi?id=17684
         if (e2x.op == TOK.type)
-            e2x = resolveAliasThis(sc, e2x);
-
-        e2x = resolveProperties(sc, e2x);
+        {
+            Expression[] results;
+            iterateAliasThis(sc, e2x, &EmptyAliasThisCtx().castToBool, results);
+            Expression ret = enforceOneResult(results, e2x.loc, "unable to represent %s as bool; candidates:", e2x.toChars());
```

Comment:
Again, I'm seeing code duplication here.  Is there a way to reduce that and supply some accompanying documentation for it?